backpack/extensions/saved_quantities.py

```diff
@@ -0,0 +1,46 @@
+"""Class for saving backpropagation quantities."""
+from typing import Dict, Union
+
+from torch import Tensor
+
+
+class SavedQuantities:
+    """Implements interface to save backpropagation quantities."""
+
+    def __init__(self):
+        """Initialization."""
+        self._saved_quantities: Dict[int, Tensor] = {}
+
+    def save_quantity(self, key: int, quantity: Tensor) -> None:
+        """Saves the quantity under the specified key.
+
+        Args:
+            key: id of grad_input[0]
+            quantity: tensor to save
+
+        Raises:
+            NotImplementedError: if the key already exists
+        """
+        if key in self._saved_quantities:
```

Comment:
`if key in self._saved_quantities.keys()`?

Reply:
I thought so, too. But [this](https://stackoverflow.com/questions/44035287/check-key-exist-in-python-dict/44035382) seems to be the correct way.

Reply:
Okay.